src/gsi/clw_mod.f90

```diff
@@ -983,6 +983,12 @@ subroutine retrieval_amsr2(tb,nchanl,clw,kraintype,ierr)
 
 
 ! ---------- Calculate predictors ---------------------------
+  if(.not.( (tb_use(7) - tb_use(8)) > 0_r_kind ).or. &
+     .not.( (tb_use(11) -tb_use(12)) > 0_r_kind ) ) then
+     write(6,*)'The issue (GSI#678) needs further investigation and be resolved if this kind of obs is used' 
+     write(6,*)'Now, GSI would abort'
+     call stop2(72)
```

Comment:
While I understand the rationale for this change (detect a protentional error), we do not want to merge this logic into `develop`.   

There are other places in `clw_mod` where a difference of two terms is used as the argument of `log`. Logic is in place to ensure the sign of the difference is greater than zero. 
 For example, lines 801 to 808 of `clw_mod.f90` in `develop` read
```
! ---------- Calculate predictors ---------------------------                                      

  if(tb_use(4) > tb_use(3)) then
     return
  endif

  pred_var_clw(1) = log(tb_use(3)-tb_use(4))  !(tb18v - tb18h)                                     
  pred_var_clw(2) = log(tb_use(6)-tb_use(7))   !(tb37v - tb37h)                                    
```
While this does not check for the possibility of a negative argument on the second `log`,  The first difference is checked.  This may be sufficient.  I am not familiar the characteristics of GMI channels.

Similar code could be placed before lines 993 and 994 and the above logic, write, and stop removed.

Reply:
@RussTreadon-NOAA  As some explanation at https://github.com/NOAA-EMC/GSI/issues/678, the problem is that we need more significant efforts to investigate the issue and the solution. For example, we don't know if 
```
if(tb_use(4) > tb_use(3)) then
     return
  endif
  ```  ( return to avoid the exception) is the one needed. It is not preferable because it won't give an warnming to the user/developer who might still think everything is going well.  That is why it is considered the current logics ( error message and stop) is needed before , in the future, when the data is to be considered and further work is applied on this issue.    That will be great if you share your opinions on that issue and more experts could discuss to decide a fix for being now.

Reply:
@RussTreadon-NOAA  "I assume the developers who wrote the code understood this and are OK with the given behavior." I am not sure if I understand what you meant.   The codes is always improving. It is obvious the "developer who wrote the code" would not be ok with the use of negative values in log function  in their  code(while it  exists to be deal with through this issue/PR) .

Reply:
@TingLei-daprediction , my comments are not meant to stop merger of this PR into GSI `develop`.  As stated above, I am not in favor of merging this type of print & abort into the authoritative GSI `develop`.  I am neither the GSI code manager nor the Handling Reviewer for this PR.  My comments do not preclude merger.

If the team decides to retain the proposed change to `clw_mod.f90` in this PR, please update the PR description to note that this PR resolves issue #678 in addition to resolving #676.

Reply:
I agree with @RussTreadon-NOAA's opinion.  At least GSI shouldn't be stopped in this situation because in the worst situation this type of satellite data, e.g., AMSR-2,  is not been used but there are many other Satellite data are still useful.  There is also another situation in which only a few points that are not valid  in the dataset. 

Print out message in this situation might also be not acceptable. Because this will print out tons of output saying one thing.